gnat2goto/driver/tree_walk.adb

```diff
@@ -2375,6 +2453,78 @@ package body Tree_Walk is
    ---------------------------
 
    procedure Do_Object_Declaration (N : Node_Id; Block : Irep) is
+      Obj_Id : constant Symbol_Id :=
+        Intern (Unique_Name (Defining_Identifier (N)));
+   begin
+      --  First check for object declarations which are not constants
+      if not Constant_Present (N) then
+         --  Not any sort of constant.
+         --  Process non-constant object_declaration.
+         Do_Object_Declaration_Full_Declaration (N, Block);
+      elsif --  Check that this isn't a completion of a deferred constant.
+         not Global_Symbol_Table.Contains (Obj_Id)
+      then
+         --  The declaration is of constant which may be deferred.
+         declare
+            Entity : constant Entity_Id := Defining_Identifier (N);
+            --  The full view of a deferred constant is obtained
+            --  by calling the Full_View function.  As the gnat front-end
+            --  has completed semantic analysis before invoking the
+            --  gnat to goto translation all object_declarations that are
+            --  deferred constants should have a full view unless the
+            --  declaration has the pragma Import applied.
+            Full_View_Entity : constant Entity_Id := Full_View (Entity);
+
+            procedure Register_Constant_In_Symbol_Table (N : Node_Id);
+            --  Adds a dummy entry to the symbol table to register that a
+            --  constant has already been processed.
+
+            procedure Register_Constant_In_Symbol_Table (N : Node_Id) is
+               Constant_Name : constant Symbol_Id :=
+                 Intern (Unique_Name (Defining_Identifier (N)));
+               Constant_Symbol : Symbol;
+            begin
+               Constant_Symbol.Name := Constant_Name;
+               Constant_Symbol.BaseName   := Constant_Name;
+               Constant_Symbol.PrettyName := Constant_Name;
+               Constant_Symbol.SymType    := Make_Nil (Sloc (N));
```

Comment:
Shouldn't we give the symbol a type as well?

Reply:
It might be cleaner but not necessary as it the symbol table content is only used to check if the name has been encountered before.